### PR TITLE
[falco] Adjust Falco events to work with External Alerts rule

### PR DESCRIPTION
**Pull request by mjwolf**
## Proposed commit message

Adjust some values in the falco ingest pipeline to work with the default 'External Alert' detection rule in Kibana. Doing this will make setting up detection alerts on ingested Falco events easier, since users can use the existing, builtin, detection rule.

Changes are:
* Set 'message' to `rule.name` - the rule uses `message` for the alert's display name
* Add host.name and user.name ECS mappings
* Set 'event.severity' to match alert severity override. The alert rule uses '=' comparator to set severity override, so the values must match exactly the new values set here for the alert to work.

The severity values are taken from the [External Alert rule definition](https://github.com/elastic/detection-rules/blob/main/rules/promotions/external_alerts.toml).

Falco priority meaning is defined [here](https://falco.org/docs/rules/basic-elements/#priority) and Elastic severity meaning is defined [here](https://www.elastic.co/guide/en/security/current/rules-ui-create.html#rule-ui-basic-params).

The mappings used in the integration are:
Falco CRITICAL or higher -> Elastic Critical
Falco ERROR -> Elastic High
Falco WARNING or NOTICE -> Elastic Medium
Falco INFORMATIONAL or lower -> Elastic Low

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [x] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## How to test this PR locally

1. Install the integration
2. In Kibana, Security-Rules add and enable the 'External Alerts' detection rule.
3. Set up Falco and falcosidekick to send alerts to ES. See https://github.com/falcosecurity/falcosidekick/tree/master?tab=readme-ov-file#installation and https://github.com/falcosecurity/falcosidekick/blob/master/docs/outputs/elasticsearch.md

### CODE REVIEW

**Review by efd6 — COMMENTED** [review]
Can you include a reference to the origin for the numeric severity values in the PR description to be included in the commit message?

**Comment by mjwolf** [comment]
> Can you include a reference to the origin for the numeric severity values in the PR description to be included in the commit message?

I've updated the description with links to the rule and the meanings of priority/severity

File: packages/falco/data_stream/alerts/elasticsearch/ingest_pipeline/default.yml [hunk 1]
```diff
@@ -651,11 +658,7 @@ processors:
 - set:
     field: orchestrator.resource.type
     value: 'pod'
-    if: ctx.containsKey('falco.output_fields.k8s.pod.name')
-- set:
```
Comment (mjwolf) [inline packages/falco/data_stream/alerts/elasticsearch/ingest_pipeline/default.yml:655]: I'm not sure why this was ever here, it looks wrong, and 'pod' resource type supported by falco: https://falco.org/docs/reference/rules/supported-fields/#field-class-k8s

File: packages/falco/data_stream/alerts/elasticsearch/ingest_pipeline/default.yml [hunk 2]
```diff
@@ -266,21 +266,21 @@ processors:
...
         } else if (priority == "notice") {
-            ctx['event.severity'] = 2;
+            ctx['event.severity'] = 21;
         } else if (priority == "informational") {
-            ctx['event.severity'] = 1;
+            ctx['event.severity'] = 21;
         } else if (priority == "debug") {
-            ctx['event.severity'] = 0;
+            ctx['event.severity'] = 21;
         }
```
[the remark this answers is not in the document]
Comment (mjwolf) [reply packages/falco/data_stream/alerts/elasticsearch/ingest_pipeline/default.yml]: Thanks, I've changed it to the 2nd version

**Review by efd6 — APPROVED** [review]

**Pull request merged**